be/src/vec/exec/format/parquet/vparquet_column_chunk_reader.h

```diff
@@ -84,41 +84,34 @@ class ColumnChunkReader {
     };
 
     ColumnChunkReader(io::BufferedStreamReader* reader, tparquet::ColumnChunk* column_chunk,
-                      FieldSchema* field_schema, cctz::time_zone* ctz, io::IOContext* io_ctx);
+                      FieldSchema* field_schema, const tparquet::OffsetIndex* offset_index,
+                      cctz::time_zone* ctz, io::IOContext* io_ctx);
     ~ColumnChunkReader() = default;
 
     // Initialize chunk reader, will generate the decoder and codec.
     Status init();
 
     // Whether the chunk reader has a more page to read.
-    bool has_next_page() { return _chunk_parsed_values < _metadata.num_values; }
-
-    // Deprecated
-    // Seek to the specific page, page_header_offset must be the start offset of the page header.
-    // _end_offset may exceed the actual data area, so we can only use the number of parsed values
-    // to determine whether there are remaining pages to read. That's to say we can't use the
-    // PageLocation in parquet metadata to seek to the specified page. We should call next_page()
-    // and skip_page() to skip pages one by one.
-    // todo: change this interface to seek_to_page(int64_t page_header_offset, size_t num_parsed_values)
-    // and set _chunk_parsed_values = num_parsed_values
-    // [[deprecated]]
-    void seek_to_page(int64_t page_header_offset) {
-        _remaining_num_values = 0;
-        _page_reader->seek_to_page(page_header_offset);
-        _state = INITIALIZED;
-    }
+    bool has_next_page() const { return _chunk_parsed_values < _metadata.num_values; }
 
     // Seek to next page. Only read and parse the page header.
     Status next_page();
 
     // Skip current page(will not read and parse) if the page is filtered by predicates.
     Status skip_page() {
         Status res = Status::OK();
-        _remaining_num_values = 0;
-        if (_state == HEADER_PARSED) {
-            res = _page_reader->skip_page();
+        if (_offset_index) {
+            _page_index++;
+            if (_page_index == _offset_index->page_locations.size()) {
+                return Status::EndOfFile("End of file");
```

Comment:
Return other errors, don't return `EndOfFile`, because is has special meaning in `VfileScanner`.